src/datasets/open_weiboscope.py

```diff
@@ -0,0 +1,185 @@
+    assert ow_files is not []
+    
+    # strategy: randomize order of weeks (individual files), sample in order from each week.
+    try:
+        random.seed(rng_seed)
+    except:
+        pass
+    random.shuffle(ow_files)
+    split_on = int(len(ow_files) * train_pct)
+    data_sets = {}
+    logger.debug("Shuffle order: {}, split on {}".format(ow_files, split_on))
+    data_sets['train'], data_sets['test'] = ow_files[:split_on], ow_files[split_on:]
+    logger.debug(data_sets)
+    for table_path in data_sets[which_set]:
+        with codecs.open(table_path, "r", encoding="utf-8") as f:
+            logging.debug("In file {}".format(table_path))
+            for line in unicode_csv_reader(f):
+                try:
+                    records_split = line
+                    if len(records_split) != 11:
+                        raise  BadRecordException("Comma split error on mid={} in"
+                                         "file {} (len of record: {})".format(
+                                            records_split[0], 
+                                            os.path.basename(table_path),
+                                            len(records_split)))
+            
+                    # text on field 6, deleted on field 9. screen out retweets (field 1) 
+                    if records_split[1] == '':
+                        if form=='hanzi':
+                            record_txt, sentiment = enforce_length(
+                                records_split[6], min_length, max_length, 
+                                pad_out), records_split[9] != '' 
+                            yield record_txt, sentiment
+                        elif form=='pinyin':
+                            record_txt, sentiment = enforce_length(
+                                romanize_tweet(records_split[6]), min_length, 
+                                max_length, pad_out), records_split[9] != '' 
+                            yield record_txt, sentiment
+                        elif form=='onehot':
+                            record_txt, sentiment = enforce_length(
+                                romanize_tweet(records_split[6]), min_length, 
+                                max_length, pad_out), records_split[9] != '' 
+                            yield text_to_one_hot(record_txt, vocabulary), sentiment
+                        else:
+                            raise Exception("Unknown form '{}' (should be 'hanzi', "
+                                            "'pinyin', or 'onehot')".format(form))
+
+                except TextTooShortException:
+                    continue
+                except BadRecordException:
+                    continue
+                except IndexError:
+                    continue
+                except UnicodeEncodeError:
+                    continue
+
+                except GeneratorExit:
```

Comment:
It might be a good idea to at least logger.info these exceptions in case someone wants to inspect logs